nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -595,8 +596,14 @@ impl DataStore {
             .filter(service_network_interface::time_deleted.is_null())
             .select(Sled::as_select());
 
-        instance_query
-            .union(service_query)
+        let sleds_filter =
+            sleds_filter.map(|sleds| sleds.to_vec()).unwrap_or(vec![]);
+        sled::table
+            // Grab just the sleds that were asked for (if any)
+            .filter(sled::id.eq_any(sleds_filter))
+            .or_filter(sled::time_deleted.is_null())
```

Comment:
Why use `.or_filter` instead of just `.filter` here? Does this matter in the case when `sleds_filter` is empty?

Reply:
Yes, `eq_any` on an empty array would return no rows. From the [diesel docs](https://docs.diesel.rs/master/diesel/expression_methods/trait.ExpressionMethods.html#example-2):
>// Calling `eq_any` with an empty array is the same as doing `WHERE 1=0`

But I think this is still not exactly what I want here. The `or_filter` basically just includes everything regardless. :/ Looks like I'll have to try going back to only adding the filter conditionally. Time to get back to wrestling with diesel...

Reply:
As a cheap fall-back plan: if you box the query, it should be easy to add conditional filters during construction:

```rust
let query = sled::table.filter(sled::time_deleted.is_null()).into_boxed();

// If no filter is provided, we send this to all sleds.
if !sleds_filter.empty() {
  query = query.filter(sled::id.eq_any(sleds_filter));
}
// Keep building query...
```

Reply:
yea, i had something like that earlier but was drowning huge errors about missing trait bounds. But that might've been due to just trying it on the union expr, i think what you have should work.